Sources/Sentry/SentryANRTracker.m

```diff
@@ -0,0 +1,110 @@
+#import "SentryANRTracker.h"
+#import "SentryCrashAdapter.h"
+#import "SentryDispatchQueueWrapper.h"
+#import "SentryLog.h"
+#import "SentryThreadWrapper.h"
+#import <Foundation/Foundation.h>
+
+@interface
+SentryANRTracker ()
+
+@property (weak, nonatomic) id<SentryANRTrackerDelegate> delegate;
+@property (nonatomic, assign) NSTimeInterval timeoutInterval;
+@property (nonatomic, strong) id<SentryCurrentDateProvider> currentDate;
+@property (nonatomic, strong) SentryCrashAdapter *crashAdapter;
+@property (nonatomic, strong) SentryDispatchQueueWrapper *dispatchQueueWrapper;
+@property (nonatomic, strong) SentryThreadWrapper *threadWrapper;
+
+@property (weak, nonatomic) NSThread *thread;
+
+@end
+
+@implementation SentryANRTracker
+
+- (instancetype)initWithDelegate:(id<SentryANRTrackerDelegate>)delegate
+           timeoutIntervalMillis:(NSUInteger)timeoutIntervalMillis
+             currentDateProvider:(id<SentryCurrentDateProvider>)currentDateProvider
+                    crashAdapter:(SentryCrashAdapter *)crashAdapter
+            dispatchQueueWrapper:(SentryDispatchQueueWrapper *)dispatchQueueWrapper
+                   threadWrapper:(SentryThreadWrapper *)threadWrapper
```

Comment:
nit: Should [we pick either `Adapter` or `Wrapper` as naming convention?](https://en.wikipedia.org/wiki/Adapter_pattern) Sounds like `Wrapper` is being introduced, so we could just rename those to `Adapter`?

Reply:
Good point. We use `wrapper` more often than `adapter`. I'm going to rename `SentryCrashAdapter` to `SentryCrashWrapper` as it's actually a wrapper around SentryCrash.

Reply:
PR is open https://github.com/getsentry/sentry-cocoa/pull/1699